src/ConditionManager.cpp

```diff
@@ -0,0 +1,87 @@
+#include <thread>
+#include <chrono>
+#include <algorithm>
+#include <exception>
+
+#include "ConditionManager.h"
+
+#include "VmeUsbBridge.h"
+#include "HV.h"
+
+// Static
+const std::vector< std::pair<ConditionManager::State, ConditionManager::State> > ConditionManager::m_transitions = {
+    { ConditionManager::State::idle, ConditionManager::State::configured },
+    { ConditionManager::State::configured, ConditionManager::State::running },
+    { ConditionManager::State::running, ConditionManager::State::idle },
+    { ConditionManager::State::configured, ConditionManager::State::idle }
+};
+
+// Static
+std::string ConditionManager::stateToString(ConditionManager::State state) {
+    switch(state) {
+        case State::idle:
+            return "idle";
+        case State::configured:
+            return "configured";
+        case State::running:
+            return "running";
+        default:
+            return "Unknown state!";
+    }
+}
+
+// Static
+bool ConditionManager::checkTransition(ConditionManager::State state_from, ConditionManager::State state_to) { 
+    return std::find(m_transitions.begin(), m_transitions.end(), std::pair<State, State>( { state_from, state_to } ) ) != m_transitions.end();
+}
+
+bool ConditionManager::setState(ConditionManager::State state) {
+    if( state == m_state ) {
+        std::cout << "ConditionManager is already in state " << stateToString(m_state) << std::endl;
+        return false;
+    }
+   
+    if( !checkTransition(m_state, state) ) { 
+        throw std::runtime_error("Transition from " + stateToString(m_state) + " to " + stateToString(state) + " is not allowed.");
+    }
+    
+    std::cout << "Changing ConditionManager state from " << stateToString(m_state) << " to " << stateToString(state) << std::endl;
+
+    m_state = state;
+
+    return true;
+}
+
+void ConditionManager::setHVPMTValue(int HVNumber, int HVValue) {
+    m_hvpmt_setValues[HVNumber] = HVValue;
```

Comment:
I would prefer the `.at()` method since it checks if the index is within the size of the vector...